internal/gitinterface/hash.go

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package gitinterface
+
+import (
+	"encoding/hex"
+	"errors"
+)
+
+const (
+	zeroSHA1HashString   = "0000000000000000000000000000000000000000"
+	zeroSHA256HashString = "0000000000000000000000000000000000000000000000000000000000000000"
+)
+
+var (
+	ErrInvalidHashEncoding = errors.New("hash string is not hex encoded")
+	ErrInvalidHashLength   = errors.New("hash string is wrong length")
+)
+
+// Hash represents a Git object hash. It is a lightweight wrapper around the
+// standard hex encoded representation of a SHA-1 or SHA-256 hash used by Git.
+type Hash struct {
+	hash string
+}
+
+// String returns the hex encoded hash.
+func (h Hash) String() string {
+	return h.hash
+}
+
+// IsZero compares the hash to see if it's the zero hash for either SHA-1 or
+// SHA-256.
+func (h Hash) IsZero() bool {
+	return h.hash == zeroSHA1HashString || h.hash == zeroSHA256HashString
+}
+
+// ZeroHash represents an empty Hash.
+// TODO: use SHA-256 zero hash for repositories that have that as the default.
+var ZeroHash = Hash{hash: zeroSHA1HashString}
```

Comment:
If you use []byte as the base storage, I think this becomes much simpler, since zero sha1 and zero sha256 should effectively both be the same, or differ only by the len of 0 bytes.

Reply:
you'll also save a ton of space if you expect a lot of hashes because of string encoding.

Reply:
Done!